ports/mimxrt/sdcard.c

```diff
@@ -0,0 +1,989 @@
+    (SDMMC_MASK(SDCARD_STATUS_LOCK_UNLOCK_FAILED_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_COM_CRC_ERROR_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_ILLEGAL_COMMAND_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_CARD_ECC_FAILED_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_CC_ERROR_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_ERROR_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_CSD_OVERWRITE_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_WP_ERASE_SKIP_SHIFT)) | \
+    (SDMMC_MASK(SDCARD_STATUS_AUTH_SEQ_ERR_SHIFT))
+
+#define SDMMC_R1_CURRENT_STATE(x) (((x) & 0x00001E00U) >> 9U)
+
+
+// ---
+// SD Card command identifiers
+// ---
+enum
+{
+    SDCARD_CMD_GO_IDLE_STATE        = 0U,
+    SDCARD_CMD_ALL_SEND_CID         = 2U,
+    SDCARD_CMD_SEND_REL_ADDR        = 3U,
+    SDCARD_CMD_SET_DSR              = 4U,
+    SDCARD_CMD_SELECT_CARD          = 7U,
+    SDCARD_CMD_SEND_IF_COND         = 8U,
+    SDCARD_CMD_SEND_CSD             = 9U,
+    SDCARD_CMD_SEND_CID             = 10U,
+    SDCARD_CMD_STOP_TRANSMISSION    = 12U,
+    SDCARD_CMD_SEND_STATUS          = 13U,
+    SDCARD_CMD_GO_INACTIVE_STATE    = 15U,
+    SDCARD_CMD_SET_BLOCKLENGTH      = 16U,
+    SDCARD_CMD_READ_SINGLE_BLOCK    = 17U,
+    SDCARD_CMD_READ_MULTIPLE_BLOCK  = 18U,
+    SDCARD_CMD_SET_BLOCK_COUNT      = 23U,
+    SDCARD_CMD_WRITE_SINGLE_BLOCK   = 24U,
+    SDCARD_CMD_WRITE_MULTIPLE_BLOCK = 25U,
+    SDCARD_CMD_PROGRAM_CSD          = 27U,
+    SDCARD_CMD_SET_WRITE_PROTECT    = 28U,
+    SDCARD_CMD_CLEAR_WRITE_PROTECT  = 29U,
+    SDCARD_CMD_SEND_WRITE_PROTECT   = 30U,
+    SDCARD_CMD_ERASE                = 38U,
+    SDCARD_CMD_LOCK_UNLOCK          = 42U,
+    SDCARD_CMD_APP_CMD              = 55U,
+    SDCARD_CMD_GEN_CMD              = 56U,
+    SDCARD_CMD_READ_OCR             = 58U,
+};
+
+// ---
+// SD Card application command identifiers
+// ---
+enum
+{
+    SDCARD_ACMD_SET_BUS_WIDTH       =  6U,
+    SDCARD_ACMD_SD_SEND_OP_COND     = 41U,
+};
+
+
+// ---
+// SD Card state identifiers
+// ---
+enum
+{
+    SDCARD_STATE_IDLE        = 0U,
+    SDCARD_STATE_READY       = 1U,
+    SDCARD_STATE_IDENTIFY    = 2U,
+    SDCARD_STATE_STANDBY     = 3U,
+    SDCARD_STATE_TRANSFER    = 4U,
+    SDCARD_STATE_SENDDATA    = 5U,
+    SDCARD_STATE_RECEIVEDATA = 6U,
+    SDCARD_STATE_PROGRAM     = 7U,
+    SDCARD_STATE_DISCONNECT  = 8U,
+};
+
+// ---
+// SD Card transfer status
+// ---
+typedef enum
+{
+    SDCARD_TRANSFER_SUCCESS = 0,
+    SDCARD_TRANSFER_ERROR,
+    SDCARD_TRANSFER_PENDING
+} sdcard_transfer_status_t;
+
+
+// ---
+// SD Card type definitions
+// ---
+typedef struct _cid_t {
+    uint8_t reserved_0;
+    uint16_t mdt : 12;
+    uint16_t reserved_1 : 4;
+    uint32_t psn;
+    uint8_t prv;
+    char pnm[6];
+    uint16_t oid;
+    uint8_t mid;
+} __attribute__((packed)) cid_t;
+
+typedef struct _csd_t {
+    uint32_t data[4];
+} __attribute__((packed)) csd_t;
+
+typedef struct _csr_t {
+    uint32_t data[2];
+} __attribute__((packed)) csr_t;
+
+
+
```

Comment:
please don't have multiple blank lines, ideally there would only be one blank line separating sections

Reply:
fixed.